src/other/broadcasting.jl

```diff
@@ -0,0 +1,63 @@
+struct LazyNewColDataFrame
+    df::DataFrame
+    col::Symbol
+end
+
+# we allow LazyNewColDataFrame only for data frames with some columns
+Base.axes(x::LazyNewColDataFrame) = axes(x.df[1])
+
+Base.Broadcast.broadcastable(df::AbstractDataFrame) = Matrix(df)
```

Comment:
Do we really need to do this? That doesn't sound like the most useful interpretation to me: `f.(df)` could also return a data frame. That would preserve column names.

Reply:
We can only return an object that supports `eachindex`. We could add this support to `DataFrame` alternatively (but I thought it is not safe to provide `eachindex` for a data frame that would iterate all cell indices, but I am not 100% sure about it). Do you prefer this?

However, bear in mind that the column names will not be retained anyway as in broadcasting access is provided via `eachindex` framework.

In short - a data frame in broadcasting must be considered as a matrix-like object (because `axes` on a data frame returns two dimensions). We can either perform the conversion or add `eachindex`. A third option is not to add this support at all and disallow data frame in RHS of broadcasting. What would you prefer?

Finally note that currently `DataFrameRow` is collected to a `Vector` when used in LHS of broadcasting, so casting a data frame to a `Matrix` felt like a consistent approach.

Reply:
OK. What's the best way to ensure `f.(df)` returns a data frame? I guess we could implement `eachindex` if that's needed.

I see the case of `DataFrame` as similar to `NamedArray` or `AxisArray`: we should preserve the type and column names if possible.

Reply:
Good point. I think I will leave out LHS broadcasting of data frame and `DataFrameRow` from this PR to a separate PR as these functionalities are independent from the core of this PR. Thank you for digging into this.

Reply:
I have opened #1806.